wfe2/wfe.go

```diff
@@ -2258,6 +2270,13 @@ func (wfe *WebFrontEndImpl) NewOrder(
 			return
 		}
 		names[i] = ident.Value
+		lowerValue := strings.ToLower(ident.Value)
+		if slices.Contains(names, lowerValue) {
+			identMap[lowerValue] = &sapb.Identifier{
+				Type:  string(ident.Type),
+				Value: ident.Value,
+			}
+		}
```

Comment:
I don't understand what this code is doing. It seems like identMap will contain a key for every all-lowercase identifier in the request.

Below, what we need is a slice of identifier objects. I think it would make more sense to skip the identMap construction, and down at line 2299 unconditionally construct the slice by appending a `*sapb.Identifier` for each name. At that point the names will already have gone through the UniqueLowerNames function. And it has the advantage that we are constructing the SA-specific type closer to where it is used in an RPC.

Reply:
Whoops, you're right. I moved some stuff around here to try and take advantage of the loop we were already in and the result doesn't make sense.